services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -150,13 +166,118 @@ class PortHistorySpec extends HealthCheckSpecification {
+        Wrappers.wait(WAIT_OFFSET) {
+            Long timestampAfterDown = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterDown)) {
+                it.size() == 2
+            }
+        }
+
+        when: "Generate antiflap statistic"
+        def count = 0
+        Integer intervalBetweenPortStateManipulation = (antiflapCooldown / 10).toInteger()
+        Wrappers.timedLoop(antiflapDumpingInterval * 0.9) {
+            northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+            sleep(intervalBetweenPortStateManipulation)
+            northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+            sleep(intervalBetweenPortStateManipulation)
+            count += 1
+        }
+
+        then: "Antiflap statistic is available in port history"
+        Wrappers.wait(antiflapDumpingInterval * 0.1 + WAIT_OFFSET) {
+            Long timestampAfterStat = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterStat)) {
+                it.size() == 3
+                def antiflapStat = it[-1]
+                checkPortHistory(antiflapStat, isl.srcSwitch.dpId, isl.srcPort,
+                        PortHistoryEvent.ANTI_FLAP_PERIODIC_STATS)
+                count == antiflapStat.downCount
+                count == antiflapStat.upCount
+            }
+        }
+
+        and: "Cleanup: revert system to original state"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval + antiflapCooldown) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+    }
+
+    def "System shows antiflap statistic in the ANTI_FLAP_DEACTIVATED event when antiflap is deactivated\
+ before collecting ANTI_FLAP_PERIODIC_STATS"() {
+        assumeTrue("It can't be run when antiflap.cooldown + flap_duration > port.antiflap.stats.dumping.interval.seconds",
+                antiflapCooldown + 3 < antiflapDumpingInterval)
+        //port up/down procedure is done once in this test, so it can't take more than 3 seconds
+
+        given: "A direct link"
+        def isl = getTopology().islsForActiveSwitches.find { !it.aswitch }
+        assumeTrue("Unable to find ISL for this test", isl as boolean)
+        def timestampBefore = System.currentTimeMillis()
+
+        when: "Execute port DOWN on the src switch for activating antiflap"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+            Long timestampAfterDown = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterDown)) {
+                it.size() == 2
+            }
+        }
+
+        and: "Generate antiflap statistic"
```

Comment:
This assumes a pretty big cooldown. You did port down, then waited for isl to get down and history to appear, and only then the port up. Can we add a cooldown assumption of at least couple of seconds at the start of the test?

Reply:
added
```
 assumeTrue("At least 10 seconds should be available for changing port status", antiflapCooldown >= 10)
        // assume that portDown/portUp can take some time on hardware env(no more than 10 seconds)
```